packages/web/src/pages/upload-page/fields/ReleaseDateField.tsx

```diff
@@ -1,94 +1,323 @@
+  }, [trackReleaseDate, releaseDateType])
 
   const onSubmit = useCallback(
     (values: ReleaseDateFormValues) => {
-      setValue(values[RELEASE_DATE] ?? null)
+      if (values[RELEASE_DATE_TYPE] === ReleaseDateType.RELEASE_NOW) {
+        setTrackReleaseDate(null)
+        return
+      }
+      const releaseDateValue = values[RELEASE_DATE]
+      const releaseDateHour = parseInt(values[RELEASE_DATE_HOUR].split(':')[0])
+      const releaseDateMeridian = values[RELEASE_DATE_MERIDIAN]
+
+      const truncatedReleaseDate = moment(releaseDateValue).startOf('day')
+
+      let adjustedHours = releaseDateHour
+      if (releaseDateMeridian === 'PM' && releaseDateHour < 12) {
+        adjustedHours += 12
+      } else if (releaseDateMeridian === 'AM' && releaseDateHour === 12) {
+        adjustedHours = 0
+      }
+      const combinedDateTime = truncatedReleaseDate.add(adjustedHours, 'hours')
+
+      setTrackReleaseDate(combinedDateTime.toString() ?? null)
+      // set other fields
     },
-    [setValue]
+    [setTrackReleaseDate]
   )
 
   const renderValue = useCallback(() => {
     return (
       <SelectedValue
-        label={
-          moment(value ?? undefined)
-            .calendar()
-            .split(' at')[0]
-        }
+        label={getScheduledReleaseLabelMessage(
+          trackReleaseDate,
+          'Scheduled for '
+        )}
         icon={IconCalendar}
       >
         <input
           className={styles.input}
           name={RELEASE_DATE}
-          value={moment(value ?? undefined).format('L')}
+          value={'hello'}
           aria-readonly
           readOnly
         />
       </SelectedValue>
     )
-  }, [value])
+  }, [trackReleaseDate])
 
   return (
-    <ContextualMenu
-      label={messages.title}
-      description={messages.description}
-      icon={<IconCalendar className={styles.titleIcon} />}
-      initialValues={initialValues}
-      onSubmit={onSubmit}
-      menuFields={
-        <>
-          <Text
-            className={cn(styles.title, styles.modalHeading)}
-            variant='title'
-            size='large'
-          >
-            {messages.title}
-          </Text>
-          <Text>{messages.description}</Text>
+    <>
+      <ContextualMenu
+        label={messages.title}
+        description={messages.description}
+        icon={<IconCalendar className={styles.titleIcon} />}
+        initialValues={initialValues}
+        validationSchema={toFormikValidationSchema(timeValidationSchema)}
+        validateOnChange={false}
+        validateOnBlur={false}
+        onSubmit={onSubmit}
+        menuFields={
+          <>
+            <div className={cn(layoutStyles.col, layoutStyles.gap4)}>
+              <Text>{messages.description}</Text>
+              <RadioItems />
+            </div>
+          </>
+        }
+        renderValue={renderValue}
+      />
+    </>
+  )
+}
+
+const RadioItems = (props: any) => {
+  const [releaseDateTypeField, ,] = useField(RELEASE_DATE_TYPE)
```

Comment:
I think you don't need the extra commas here (you do need them when you want a later element such in as the next line)